python/spacewalk/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -1086,7 +1086,7 @@ def get_mediaproducts(self):
         """
         media_products_path = os.path.join(self._get_repodata_path(), 'media.1/products')
         try:
-            (s,b,p,q,f,o) = urlparse(self.url)
+            (s,b,p,q,f,o) = urlparse(self.repo.urls[0])
```

Comment:
After looking at the test case and trying to understand the bug, a question came to my mind: Is it correct to always use the first entry of the mirror list? Based on the name "mirror list", I'd expect that a client might want to pick a mirror that's not the first on the list (e.g. because that one is temporarily down).

Reply:
Hey Alex, I put some more thoughts on this and I was able to make `get_mediaproducts` and also `get_get` to actually use a mirror list if available.

Now the behavior should be as you described, iterating over mirrors in case of a problem.